### PR TITLE
Spring start

### PR DESCRIPTION
**Pull request by werbower**

### CODE REVIEW

**Comment by werbower** [comment]
майн запущен на спринге. правильно ли я подошел к концепции @Autowired?

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/RouterConfig.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.softgroup.common.router.api;
+
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Created by user on 01.03.2017.
+ */
+@Configuration
+@ComponentScan
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/RouterConfig.java:10]: @werbower Where is package path to scan?

File: softgroup/main/src/main/java/MainConfig.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import com.softgroup.common.router.api.RouterConfig;
+import com.softgroup.firstrouter.api.FirstRouterConfig;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+
+/**
+ * Created by user on 01.03.2017.
+ */
+@Configuration
+@Import({RouterConfig.class
```
Comment (Dominys) [inline softgroup/main/src/main/java/MainConfig.java:11]: @werbower What is main ?

File: softgroup/main/src/main/java/MainSpring.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by user on 01.03.2017.
+ */
+public class MainSpring {
+    public static void main(String[] args) {
```
Comment (Dominys) [inline softgroup/main/src/main/java/MainSpring.java:16]: @werbower What? Main method O_o ??

File: softgroup/services/authorization/authorization-api/src/main/java/com/softgroup/authorization/api/AuthorizationConfig.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package com.softgroup.authorization.api;
+
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Created by user on 01.03.2017.
+ */
+@Configuration
+@ComponentScan
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-api/src/main/java/com/softgroup/authorization/api/AuthorizationConfig.java:10]: @werbower Package

File: softgroup/services/firstrouter/pom.xml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>firstrouter</artifactId>
+    <packaging>pom</packaging>
+
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>com.softgroup</groupId>
+                <artifactId>authorization-api</artifactId>
```
Comment (Dominys) [inline softgroup/services/firstrouter/pom.xml:19]: @werbower Why we need auth in router ?

File: softgroup/services/firstrouter/src/main/java/com/softgroup/firstrouter/api/FirstRouterConfig.java [hunk 6]
```diff
@@ -0,0 +1,12 @@
+package com.softgroup.firstrouter.api;
+
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Created by user on 02.03.2017.
+ */
+@Configuration
+@ComponentScan
```
Comment (Dominys) [inline softgroup/services/firstrouter/src/main/java/com/softgroup/firstrouter/api/FirstRouterConfig.java:10]: @werbower Why configuration in api? Where is package ?

File: softgroup/services/firstrouter/src/main/java/com/softgroup/firstrouter/api/RequestRouter.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+import javax.annotation.PostConstruct;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by user on 28.02.2017.
+ */
+@Component
+public class RequestRouter extends AbstractRouter implements RequestHandler{
```
Comment (Dominys) [inline softgroup/services/firstrouter/src/main/java/com/softgroup/firstrouter/api/RequestRouter.java:20]: @werbower This must be in abstract class.

File: softgroup/services/authorization/authorization-api/src/main/java/com/softgroup/authorization/api/router/AuthorizationRouter.java [hunk 8]
```diff
@@ -1,33 +1,40 @@
...
+import java.util.List;
 import java.util.Map;
 
 /**
  * Created by user on 27.02.2017.
  */
+@Component
 public class AuthorizationRouter extends HandlerRouter implements RouterHandler {
+    @Autowired
+    private List<AuthorizationHandler> listHandlers;
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-api/src/main/java/com/softgroup/authorization/api/router/AuthorizationRouter.java:21]: @werbower This must be in abstract class.

**Pull request closed without merging**